revision.c

```diff
@@ -3326,6 +3399,43 @@ static void expand_topo_walk(struct rev_info *revs, struct commit *commit)
 	}
 }
 
+static void prepare_to_use_bloom_filter(struct rev_info *revs)
+{
+	int i;
+	
+	// t4202 has a test that caught this seg fault.
+	if (!revs->commits)
+	    return;
+
+	parse_commit(revs->commits->item);
+
+	if (!the_repository->objects->commit_graph)
+		return;
+
+	revs->bloom_settings = the_repository->objects->commit_graph->settings;
+	if (!revs->bloom_settings)
+		return;
+
+	for (i = 0; i < revs->pruning.pathspec.nr; i++) {
+		struct pathspec_item *pi = &revs->pruning.pathspec.items[i];
+		const char *path = pi->match;
+		size_t len = strlen(path);
+
+		revs->bloom_key = xmalloc(sizeof(struct bloom_key));
+		fill_bloom_key(path, len, revs->bloom_key, revs->bloom_settings);
+
+		/* TODO: handle multiple pathspecs */		
+		break;
+	}
+
+	if (1 || trace_want(&trace_bloom_filter)) {
```

Comment:
Remove `1 ||`? Or just remove this `if` as we can use the `trace2` logic anyway to control the tracing level.